src/Azure/Orleans.Reminders.AzureStorage/Storage/AzureBasedReminderTable.cs

```diff
@@ -116,15 +140,25 @@ private static ReminderTableEntry ConvertToTableEntry(ReminderEntry remEntry, st
             };
         }
 
-        public Task TestOnlyClearTable()
+        public async Task TestOnlyClearTable()
         {
-            return this.remTableManager.DeleteTableEntries();
+            if (!_initializationTask.Task.IsCompleted)
```

Comment:
with the proposed change above, i think we should just call `await _initializationTask.Task; `all the time. That way if it's canceled or failed, it will bubble up the exception from any read/write call.

Reply:
Done